library/Fission/Internal/Orphanage/ToServerError.hs

```diff
@@ -0,0 +1,47 @@
+{-# OPTIONS_GHC -fno-warn-orphans #-}
+
+module Fission.Internal.Orphanage.ToServerError () where
+
+import Fission.Prelude
+import Network.IPFS.ToServerError
+import Servant.Server
+
+instance ToServerError Int where
+  toServerError = \case
+    300 -> err300
+    301 -> err301
+    302 -> err302
+    303 -> err303
+    304 -> err304
+    305 -> err305
+    307 -> err307
+
+    400 -> err400
+    401 -> err401
+    402 -> err402
+    403 -> err403
+    404 -> err404
+    405 -> err405
+    406 -> err406
+    407 -> err407
+    409 -> err409
+    410 -> err410
+    411 -> err411
+    412 -> err412
+    413 -> err413
+    414 -> err414
+    415 -> err415
+    416 -> err416
+    417 -> err417
+    418 -> err418
+    422 -> err422
+
+    500 -> err500
+    501 -> err501
+    502 -> err502
+    503 -> err503
+    504 -> err504
+    505 -> err505
+
+    n -> error <| show n <> " is not an error status code"
+
```

Comment:
Wait, why is this an orphan?

Reply:
I see. I moved to the helper lib 🤔

[TL;DR think "separation of concerns"]

I would personally have the class live in the application, not in the IPFS lib. The IPFS lib needs to know which error to throw, but doesn't need a generic method to convert into HTTP errors, right? If you need to set IPFS errors to Servant errors, we can do that in the module where `ToServantError` is defined.

Am I missing something?

Reply:
yeah that makes sense.

Would it be bad practice to put all the `ToServantError` orphans in the module where the class is defined?

Reply:
# Short Answer

It's not an orphan if you define it in the same module as the `class`. Let's do that 👍

# Long Answer

"Orphans" only refer to a specific edge case. By default, Haskell wants you to define `instance`s _either_ where the `data` is defined, or where the `class` is defined.

Orphans arise when you have data from a library that you don't control, and a class from a library that you don't control, and you want to implement them together without wrapping in a `newtype`.

Reply:
ahhh got it. I thought it was specifically where the `data` was defined. Nice that makes sense :+1:

Reply:
FYI, rounded out that answer a bit more